alpakka/src/main/scala/org/scanamo/ops/retrypolicy/RetryPolicy.scala

```diff
@@ -14,11 +15,16 @@ sealed abstract class RetryPolicy extends Product with Serializable { self =>
   }
 
   final def delay: Long = self match {
-    case Constant(retryDelay)       => retryDelay.toMillis
-    case Linear(retryDelay, factor) => retryDelay.toMillis * factor.toLong
+    case Constant(retryDelay) =>
```

Comment:
How about adding a `case class Jitter(min: Double, max: Double, policy: RetryPolicy)` so that you can just add a case:
```scala
case Jitter(min, max, policy) => getDelayWithJitter(min, max, policy.delay)
```

Reply:
@regiskuckaertz Pattern matching is required to handle cases like:-

1. Constant:- `retryDelay.toMillis`
2. Linear:- `retryDelay.toMillis * factor.toLong`
3. Exponential:- `delayMillis * Math.pow(delayMillis.toDouble, factor).toLong`

Reply:
Yes of course! The thing is, jitter would be a property that you add to an existing policy, instead of it being latent, so:
```scala
case class Jitter(random: SplittableRandom, margin: Long, inner: RetryPolicy) extends RetryPolicy

def delay: Long = self match {
  case Constant(x) => x
  ...
  case Jitter(rand, margin, inner) =>
    // recursively calculate the delay
    val delay = inner.delay
    val (lo, hi) = (Math.max(0, delay - margin), delay + margin + 1)
    rand.nextLong(lo, hi)
```